src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,279 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement
+                        = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            int rowsUpdated = preparedStatement.executeUpdate();
+            return rowsUpdated > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error deleting car: " + "by id = " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query
+                = "SELECT c.id, c.model, "
+                + "m.id as manufacturer_id, m.name as manufacturer_name "
+                + "FROM cars c "
+                + "JOIN cars_drivers cd ON c.id = cd.car_id "
+                + "JOIN drivers d ON cd.driver_id = d.id "
+                + "JOIN manufacturers m ON c.manufacturers_id = m.id "
+                + "WHERE d.id = ? AND c.is_deleted = false";
+        List<Car> cars;
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement
+                        = connection
+                        .prepareStatement(query)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            cars = getListCars(resultSet);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error retrieving cars by driverID: " + driverId, e);
+        }
+        return cars;
+    }
+
+    private List<Driver> getDriversForCar(Long carId) throws SQLException {
+        String query
+                = "SELECT d.id, d.name, d.license_number "
+                + "FROM drivers d "
+                + "JOIN cars_drivers cd ON d.id = cd.driver_id "
+                + "WHERE cd.car_id = ? AND d.is_deleted = false";
+        List<Driver> drivers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement
+                        = connection
+                        .prepareStatement(query)) {
+            preparedStatement.setLong(1, carId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            while (resultSet.next()) {
+                drivers.add(getDriver(resultSet));
+            }
+        }
+        return drivers;
+    }
+
+    private Driver getDriver(ResultSet resultSet) throws SQLException {
+        Long driverId = resultSet.getLong("id");
+        String name = resultSet.getString("name");
+        String licenseNumber = resultSet.getString("license_number");
+        return new Driver(driverId, name, licenseNumber);
+    }
+
+    private void createCarsDriversRelations(Car car, Connection connection) throws SQLException {
+        for (Driver driver : car.getDrivers()) {
+            String query = "SELECT * FROM drivers WHERE name = ? AND license_number = ?";
```

Comment:
Why do you need SELECT  FROM drivers at all? Everything you need you already have inside **car** object
I see you have `addDriversToCar` - use it and delete this method)